src/components/record/related-table-actions.tsx

```diff
@@ -496,50 +603,293 @@ const RelatedTableActions = ({
+   */
+  const renderButton = (button: any, index: number) => {
+    const buttonText = button.textContent;
+    const spanClass = button.children[0].className;
+    const createBtnTooltip = renderCreateBtnTooltip();
+    const unlinkBtnTooltip = (
+      <span>
+        Disconnect {currentTable} records from this {mainTable}.
+      </span>
+    );
+    const exploreBtnTooltip = (
+      <span>
+        Explore more {currentTable} records related to this {mainTable}.
+      </span>
+    );
+    let tooltip: any;
+    switch (buttonText) {
+      case 'Explore':
+        tooltip = exploreBtnTooltip;
+        return (
+          <ChaiseTooltip
+            placement='top'
+            tooltip={
+              <span>
+                Explore more {currentTable} records related to this {mainTable}.
+              </span>
+            }
+          >
+            <a
+              className='chaise-btn chaise-btn-tertiary more-results-link dropdown-button'
+              href={exploreLink}
+              onClick={onExplore}
+            >
+              <span className='chaise-btn-icon fa-solid fa-magnifying-glass'></span>
+              <span>Explore</span>
+            </a>
+          </ChaiseTooltip>
+        );
+      case 'Bulk Edit':
+        tooltip = exploreBtnTooltip;
+        return (
+          <ChaiseTooltip
+            placement='top'
+            tooltip={
+              <span>
+                Explore more {currentTable} records related to this {mainTable}.
+              </span>
+            }
+          >
+            <a
+              className='chaise-btn chaise-btn-tertiary more-results-link dropdown-button'
+              href={exploreLink}
+              onClick={onExplore}
+            >
+              <span className='chaise-btn-icon fa fa-pencil'></span>
+              <span>Bulk Edit</span>
+            </a>
+          </ChaiseTooltip>
+        );
+      case 'Link Records':
+      case 'Add Records':
+        tooltip = createBtnTooltip;
+        break;
+      case 'Unlink records':
+        tooltip = unlinkBtnTooltip;
+        break;
+      case 'Edit mode':
+      case 'Custom mode':
+      case 'Table mode':
+        return renderCustomModeBtn(true);
+    }
+
+    return (
+      <ChaiseTooltip placement='top' tooltip={tooltip}>
+        <button
+          key={`button-${index}`}
+          className={'chaise-btn chaise-btn-tertiary dropdown-button'}
+        >
+          <span className={spanClass}></span>
+          <span className='dropdown-button-text'>{buttonText}</span>
+        </button>
+      </ChaiseTooltip>
+    );
+  };
+  /*
+   * This hook is to push the buttons to the dropdown if it exceeds the container width.
+   * We are using ResizeSensor to listen to the resize event.
+   */
+  useLayoutEffect(() => {
+    const mainnav = document.querySelector('#navheader') as HTMLElement;
+    if (!buttonRef.current) return;
+
+    const calculateButtons = () => {
+      setIsMobile(window.innerWidth < 600);
```

Comment:
It doesn't makes sense to have a ResizeSensor and also something special based on the window width. Remember that window width doesn't mean much in this case, as you can resize the side panel (toc) without changing the window width.

So it should be based on the width of your actual container, not the whole window.